libpod/container_internal_common.go

```diff
@@ -374,6 +374,27 @@ func (c *Container) generateSpec(ctx context.Context) (s *spec.Spec, cleanupFunc
 				// Podman decided for --no-dereference as many
 				// bin-utils tools (e..g, touch, chown, cp) do.
 				options = append(options, "copy-symlink")
+			// TODO: this also ends up checking non-user mounts
+			case "ro", "rro":
```

Comment:
Why check `rro` here? It seems like we should always pass it along unmodified, even if unsupported - if the user explicitly requested `rro` and we silently downgrade, that seems like a problem.

Reply:
yes, rro must be passed as it is.

Also we need to have a way to restore the previous behavior.  If a user really wants a `ro`, there is no way to achieve it when we always force `ro` -> `rro`

Reply:
> Why check `rro` here? It seems like we should always pass it along unmodified, even if unsupported - if the user explicitly requested `rro` and we silently downgrade, that seems like a problem.

Downgrade shouldn't happen. The intent for an `rro` check here is to _upgrade_ `ro` to `rro` when supported _and_ to error out when a user explicitly requests `rro` and it isn't supported.

Reply:
> Also we need to have a way to restore the previous behavior. If a user really wants a `ro`, there is no way to achieve it when we always force `ro` -> `rro`

But why? I can understand for backwards compat but that _is_ wrong from a security point of view. I don't know how bad the impact would be when we roll out this change, but we can introduce a non-recursive (`norro`) as part of the transition till the buggy `ro` behaviour fades out. Wdyt?